data/data/agent/files/usr/local/bin/release-image.sh.template

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+# This library provides an `image_for` helper function which can get the
+# pull spec for a specific image in a release.
+
+image_for() {
+   podman run --rm --net=none "${RELEASE_IMAGE_DIGEST}" image "${1}"
+}
+
+
+ImageType=${2}
+ImagesEnvFile=${1}/agent-images.env
+
+# Check if we have already found this image
+case ${ImageType} in
+     agent-installer-api-server)
+          if grep -q "SERVICE_IMAGE" ${ImagesEnvFile}; then
+             return
+          fi
+          ;;
+     agent-installer-csr-approver)
+          if grep -q "CONTROLLER_IMAGE" ${ImagesEnvFile}; then
+             return
+          fi
+          ;;
+     agent-installer-orchestrator)
+          if grep -q "INSTALLER_IMAGE" ${ImagesEnvFile}; then
+             return
+          fi
+          ;;
+     agent-installer-node-agent)
+          if grep -q "AGENT_IMAGE" ${ImagesEnvFile}; then
+             return
+          fi
+          ;;
+esac
+
+# Convert the release image pull spec to an "absolute" form if a digest is available
+if ! RELEASE_IMAGE_DIGEST=$( podman inspect {{.ReleaseImage}} --format '{{"{{"}} index .RepoDigests 0 {{"}}"}}' ) || [[ -z "${RELEASE_IMAGE_DIGEST}" ]]; then
+	echo "Warning: Could not resolve release image to pull by digest" 2>&1
+	RELEASE_IMAGE_DIGEST="{{.ReleaseImage}}"
+fi
```

Comment:
Is it worth seeing if we can pull the file https://github.com/openshift/installer/blob/agent-installer/data/data/bootstrap/files/usr/local/bin/release-image.sh.template into our Ignition as well so that we don't have to maintain a copy of it?

Reply:
I actually tried early on that but ran into some problems with AddStorageFiles() with a file from another directory. For now I have just copied release-image.sh to data/data/agent, but can revisit just using the one in bootstrap if I can get it to work.

Reply:
Changed to pull the file from bootstrap and removed the copy in agent.